cartography/intel/azure/util/credentials.py

```diff
@@ -0,0 +1,173 @@
+import logging
+from datetime import datetime
+from datetime import timedelta
+from typing import Any
+from typing import Optional
+
+import adal
+import requests
+from azure.common.credentials import get_azure_cli_credentials
+from azure.common.credentials import get_cli_profile
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.core.exceptions import HttpResponseError
+from msrestazure.azure_active_directory import AADTokenCredentials
+
+
+logger = logging.getLogger(__name__)
+AUTHORITY_HOST_URI = 'https://login.microsoftonline.com'
+
+
+class Authenticator:
+
+    def authenticate_cli(self) -> Any:
+        """
+        Implements authentication for the Azure provider
+        """
+        try:
+
+            # Set logging level to error for libraries as otherwise generates a lot of warnings
+            logging.getLogger('adal-python').setLevel(logging.ERROR)
+            logging.getLogger('msrest').setLevel(logging.ERROR)
+            logging.getLogger('msrestazure.azure_active_directory').setLevel(logging.ERROR)
+            logging.getLogger('urllib3').setLevel(logging.ERROR)
+
+            arm_credentials, subscription_id, tenant_id = get_azure_cli_credentials(with_tenant=True)
+            aad_graph_credentials, placeholder_1, placeholder_2 = get_azure_cli_credentials(
+                with_tenant=True, resource='https://graph.windows.net',
+            )
+
+            profile = get_cli_profile()
+
+            return Credentials(
+                arm_credentials, aad_graph_credentials, tenant_id=tenant_id,
+                current_user=profile.get_current_account_user(), subscription_id=subscription_id,
+            )
+
+        except HttpResponseError as e:
+            if ', AdalError: Unsupported wstrust endpoint version. ' \
+                    'Current support version is wstrust2005 or wstrust13.' in e.args:
```

Comment:
Nit: s/Current support version/Current supported version/

Reply:
fixed in latest version